lib/active_record/virtual_attributes.rb

```diff
@@ -56,13 +56,23 @@ def virtual_column(name, type:, **options)
         virtual_attribute(name, type, **options)
       end
 
-      def virtual_attribute(name, type, uses: nil, arel: nil, **options)
+      def virtual_attribute(name, type, through: nil, uses: through, arel: nil, source: name, default: nil, **options)
         name = name.to_s
         reload_schema_from_cache
 
         self.virtual_attributes_to_define =
           virtual_attributes_to_define.merge(name => [type, options])
 
+        if through
+          define_delegate(name, source, :to => through, :allow_nil => true, :default => default)
+
+          unless (to_ref = reflection_with_virtual(through))
+            raise ArgumentError, "Delegation needs an association. Association #{to} does not exist"
+          end
+
+          arel = virtual_delegate_arel(source, to_ref)
```

Comment:
Since this defines `arel`, if `arel` and `through` are both passed, should that be an ArgumentError?

Reply:
I'm thinking passing `arel` should be allowed, though I agree it defeats the purpose of the `:through`

```diff
- arel = virtual_delegate_arel(source, to_ref)
+ arel ||= virtual_delegate_arel(source, to_ref)
```